src/main/java/org/commonjava/util/sidecar/jaxrs/PreSeedResource.java

```diff
@@ -0,0 +1,135 @@
+import java.util.Optional;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_OCTET_STREAM;
+import static org.commonjava.util.sidecar.services.ArchiveRetrieveService.isDecompressed;
+import static org.commonjava.util.sidecar.services.ProxyService.doProxy;
+import static org.eclipse.microprofile.openapi.annotations.enums.ParameterIn.PATH;
+
+@Path( "/api/folo/track/{id}/maven/{type: (hosted|group|remote)}/{name}" )
+public class PreSeedResource
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    ProxyService proxyService;
+
+    @Inject
+    ArchiveRetrieveService archiveService;
+
+    @HEAD
+    public Uni<Response> head( @PathParam( "path" ) String path, final @Context HttpServerRequest request )
+                    throws Exception
+    {
+        logger.debug( "Head resource: {}", path );
+        return proxyService.doHead( path, request );
+    }
+
+    @Operation( description = "Retrieve Maven artifact content from historical archive or proxy" )
+    @APIResponse( responseCode = "200", description = "Content stream" )
+    @APIResponse( responseCode = "404", description = "Content is not available" )
+    @Path( "{path: (.*)}" )
+    @Produces( APPLICATION_OCTET_STREAM )
+    @GET
+    public Uni<Response> get( @Parameter( in = PATH, required = true ) @PathParam( "id" ) final String id,
+                              @Parameter( in = PATH, schema = @Schema( enumeration = { "hosted", "group",
+                                              "remote" } ), required = true ) @PathParam( "type" ) final String type,
+                              @Parameter( in = PATH, required = true ) @PathParam( "name" ) final String name,
+                              @PathParam( "path" ) String path, final @Context HttpServerRequest request )
+                    throws Exception
+    {
+        if ( doProxy( path ) )
+        {
+            logger.debug( "Get proxy resource: {}", path );
+            return proxyService.doGet( path, type, name, request );
+        }
+        if ( !isDecompressed() )
+        {
+            boolean success = archiveService.decompressArchive();
+            if ( !success )
+            {
+                return proxyService.doGet( path, type, name, request );
+            }
+        }
+        Optional<File> download = archiveService.getLocally( path );
+        if ( download.isPresent() )
+        {
+            InputStream inputStream = FileUtils.openInputStream( download.get() );
+            final Response.ResponseBuilder builder = Response.ok( new TransferStreamingOutput( inputStream ) );
+            logger.debug( "Download path: {} from historical archive.", path );
+            return Uni.createFrom().item( builder.build() );
+        }
+        else
+        {
+            return proxyService.doGet( path, type, name, request );
+        }
+    }
+
+    @POST
+    public Uni<Response> post( @PathParam( "path" ) String path, InputStream is,
```

Comment:
Where is this "path" param defined?

Reply:
@ligangty , Same as HEAD, PUT, DELETE, just moved from original ProxyResource, don't focus such functions yet in the archive/content serving.